triedb/pathdb/history_inspect.go

```diff
@@ -50,12 +50,12 @@ func sanitizeRange(start, end uint64, freezer ethdb.AncientReader) (uint64, uint
 	if err != nil {
 		return 0, 0, err
 	}
-	last := head - 1
+	last := head
 	if end != 0 && end < last {
 		last = end
 	}
 	// Make sure the range is valid
-	if first >= last {
+	if first > last {
```

Comment:
first == last is allowed for querying single item

Reply:
What I meant is you shouldn't change the condition here.
Please revert to `first >= last`